pkg/internal/cmp/slice.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmp
+
+import "k8s.io/utils/pointer"
+
+func Equals(slice1, slice2 []*string) bool {
```

Comment:
Would we consider 2 string slices to be equal if they had the same elements but in a different order?

For example:
- "test1","test2"
- "test2", "test1"

Reply:
I was hoping to add the sort func for both the slices, but since the slices are []*string we dont have inbuilt sort func, the only way I have in mind is to convert []*string to []string and then sort and then do the above logic, do you have any other alternative in mind?

Reply:
added customized sorting